pwiz_tools/Shared/CommonUtil/SystemUtil/HttpClientWithProgress.cs

```diff
@@ -1020,9 +1125,93 @@ public static bool IsNetworkReallyAvailable()
             return false;
         }
 
+        /// <summary>
+        /// Disposes this HttpClientWithProgress instance.
+        /// Note: The underlying HttpClient is a singleton and is never disposed.
+        /// This method clears instance state but does not dispose the shared HttpClient.
+        /// </summary>
         public void Dispose()
         {
-            _httpClient?.Dispose();
+            // Clear per-request state
+            _authHeader = null;
+            _customHeaders.Clear();
+            // Note: _cookieContainer is managed by the caller (e.g., HttpPanoramaRequestHelper)
+            // and should not be disposed here. The shared HttpClient is never disposed.
+        }
+    }
+
+    /// <summary>
+    /// Provides a singleton HttpClient instance that is shared across all HttpClientWithProgress instances.
+    /// The HttpClient is created on first use and lives until process termination to avoid thread handle leaks.
+    /// </summary>
+    internal static class HttpClientSingleton
+    {
+        private static readonly Lazy<HttpClient> _instance = new Lazy<HttpClient>(CreateHttpClient, LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
I would probably make this:
`private static HttpClient _instance = CreateHttpClient();`
The "Lazy" part isn't really necessary because the first time anyone uses the class HttpClientSingleton is the same as when they first access `_instance.Value`

Reply:
Good clean-up suggestion.